validator/src/main.rs

```diff
@@ -1978,7 +1984,17 @@ pub fn main() {
 
     let private_rpc = matches.is_present("private_rpc");
     let no_port_check = matches.is_present("no_port_check");
-    let no_rocksdb_compaction = matches.is_present("no_rocksdb_compaction");
+    let rocksdb_compaction = match (
+        matches.is_present("rocksdb_compaction"),
+        matches.is_present("no_rocksdb_compaction"),
+    ) {
+        (true, true) => panic!("conflicting options are supplied!"),
```

Comment:
Use `Err`?

Reply:
You can just do this in the `Arg` definitions with [conflicts_with](https://docs.rs/clap/2.33.3/clap/struct.Arg.html#method.conflicts_with), unless I'm missing something about the desired behavior